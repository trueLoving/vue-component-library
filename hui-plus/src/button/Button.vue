<template>
  <button>
    <h1>Button HelloWorld</h1>
  </button>
</template>
