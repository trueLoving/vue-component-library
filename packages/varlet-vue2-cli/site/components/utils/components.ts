import Vue, { ComponentOptions } from 'vue'
import { CombinedVueInstance } from 'vue/types/vue'

export function pickProps<T, U extends keyof T>(props: T, propsKey: U): T[U]
export function pickProps<T, U extends keyof T>(props: T, propsKey: U[]): Pick<T, U>
export function pickProps(props: any, propsKey: any): any {
  return Array.isArray(propsKey)
    ? propsKey.reduce((pickedProps: any, key) => {
        pickedProps[key] = props[key]
        return pickedProps
      }, {})
    : props[propsKey]
}

export interface MountComponentApi {
  instance: CombinedVueInstance<any, any, any, any, any>
  unmount(): void
}

export function mountComponent(component: ComponentOptions<Vue>, container = 'body', options = {}): MountComponentApi {
  const instance = new (Vue.extend(component))(options)
  const el = instance.$mount().$el
  const wrapper = document.querySelector(container)!
  wrapper.appendChild(el)

  return {
    instance,
    unmount() {
      instance.$destroy()
      wrapper.removeChild(el)
    },
  }
}
