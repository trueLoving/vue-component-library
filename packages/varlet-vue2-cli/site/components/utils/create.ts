import Vue, { ComponentOptions, RenderContext, VueConstructor } from 'vue'
import { SlotsMixin } from './mixins/slots'
import { ListenersMixin } from './mixins/listeners'

export type ComponentOptionsPlugin = ComponentOptions<Vue> & { install?: (app: VueConstructor) => void }

export function nextTick() {
  return new Promise(Vue.nextTick)
}

export function mergeSlots(context: RenderContext) {
  const scopedSlots = context.scopedSlots ?? context.data.scopedSlots ?? {}
  const slots = context.slots()

  Object.keys(slots).forEach((key) => {
    if (!scopedSlots[key]) {
      scopedSlots[key] = () => slots[key]
    }
  })

  return scopedSlots
}

export function defineComponent(sfc: ComponentOptionsPlugin & { name: string }) {
  if (sfc.mixins) {
    sfc.mixins.push(SlotsMixin)
    sfc.mixins.push(ListenersMixin)
  } else {
    sfc.mixins = [SlotsMixin, ListenersMixin]
  }

  return sfc
}
