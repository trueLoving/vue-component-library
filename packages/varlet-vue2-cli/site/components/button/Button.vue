<template>
  <button
    v-ripple="{ disabled: disabled || !ripple }"
    class="var-site-button var--box"
    :class="[
      `var-site-button--${size}`,
      block ? 'var--flex var-site-button--block' : 'var--inline-flex',
      disabled ? 'var-site-button--disabled' : null,
      text ? `var-site-button--text-${type}` : `var-site-button--${type}`,
      text ? 'var-site-button--text' : 'var-elevation--2',
      text && disabled ? 'var-site-button--text-disabled' : null,
      round ? 'var-site-button--round' : null,
      outline ? 'var-site-button--outline' : null,
    ]"
    :style="{
      color: textColor,
      background: color,
    }"
    :disabled="disabled"
    @click="handleClick"
    @touchstart="handleTouchstart"
  >
    <var-site-loading
      class="var-site-button__loading"
      var-site-button-cover
      :type="loadingType"
      :size="loadingSize"
      :radius="loadingRadius"
      v-if="loading || pending"
    />
    <div class="var-site-button__content" :class="[loading || pending ? 'var-site-button--hidden' : null]">
      <slot />
    </div>
  </button>
</template>

<script>
import Ripple from '../ripple'
import VarSiteLoading from '../loading'
import { defineComponent } from '../utils/create'
import { props } from './props'

export default defineComponent({
  name: 'VarSiteButton',

  components: {
    VarSiteLoading,
  },

  directives: { Ripple },

  props,

  data: () => ({
    pending: false,
  }),

  methods: {
    attemptAutoLoading(result) {
      if (this.autoLoading) {
        this.pending = true
        Promise.resolve(result).finally(() => {
          this.pending = false
        })
      }
    },

    handleClick(e) {
      const { loading, disabled, pending } = this
      const { onClick } = this.getListeners()

      if (!onClick || loading || disabled || pending) {
        return
      }

      this.attemptAutoLoading(onClick(e))
    },

    handleTouchstart(e) {
      const { loading, disabled, pending } = this
      const { onTouchstart } = this.getListeners()

      if (!onTouchstart || loading || disabled || pending) {
        return
      }

      this.attemptAutoLoading(onTouchstart(e))
    },
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import '../styles/elevation';
@import '../ripple/ripple';
@import '../loading/loading';
@import './button';
</style>
