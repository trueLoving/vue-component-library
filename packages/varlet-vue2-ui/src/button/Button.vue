<template>
  <button
    v-ripple="{ disabled: disabled || !ripple }"
    class="var-button var--box"
    :class="[
      `var-button--${size}`,
      block ? 'var--flex var-button--block' : 'var--inline-flex',
      disabled ? 'var-button--disabled' : null,
      text ? `var-button--text-${type}` : `var-button--${type}`,
      text ? 'var-button--text' : 'var-elevation--2',
      text && disabled ? 'var-button--text-disabled' : null,
      round ? 'var-button--round' : null,
      outline ? 'var-button--outline' : null,
    ]"
    :style="{
      color: textColor,
      background: color,
    }"
    :disabled="disabled"
    @click="handleClick"
    @touchstart="handleTouchstart"
  >
    <var-loading
      class="var-button__loading"
      var-button-cover
      :type="loadingType"
      :size="loadingSize"
      :radius="loadingRadius"
      v-if="loading || pending"
    />
    <div class="var-button__content" :class="[loading || pending ? 'var-button--hidden' : null]">
      <slot />
    </div>
  </button>
</template>

<script>
import Ripple from '../ripple'
import VarLoading from '../loading'
import { defineComponent } from '../utils/create'
import { props } from './props'

export default defineComponent({
  name: 'VarButton',

  components: {
    VarLoading,
  },

  directives: { Ripple },

  props,

  data: () => ({
    pending: false,
  }),

  methods: {
    attemptAutoLoading(result) {
      if (this.autoLoading) {
        this.pending = true
        Promise.resolve(result).finally(() => {
          this.pending = false
        })
      }
    },

    handleClick(e) {
      const { loading, disabled, pending } = this
      const { onClick } = this.getListeners()

      if (!onClick || loading || disabled || pending) {
        return
      }

      this.attemptAutoLoading(onClick(e))
    },

    handleTouchstart(e) {
      const { loading, disabled, pending } = this
      const { onTouchstart } = this.getListeners()

      if (!onTouchstart || loading || disabled || pending) {
        return
      }

      this.attemptAutoLoading(onTouchstart(e))
    },
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import '../styles/elevation';
@import '../ripple/ripple';
@import '../loading/loading';
@import './button';
</style>
