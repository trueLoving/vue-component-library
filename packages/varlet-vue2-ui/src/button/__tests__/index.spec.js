import example from '../example'
import Vue from 'vue'
import Button from '..'
import VarButton from '../Button'
import { mount } from '@vue/test-utils'
import { delay, trigger } from '../../utils/jest'

test('test button example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test button plugin', () => {
  Vue.use(Button)
  expect(Vue.component(Button.name)).toBeTruthy()
})

test('test button onClick & onTouchstart null callback', async () => {
  const wrapper = mount(VarButton)
  await trigger(wrapper, 'click')
  await trigger(wrapper, 'touchstart')
  wrapper.destroy()
})

test('test button onClick & onTouchstart', async () => {
  const onClick = jest.fn()
  const onTouchstart = jest.fn()

  const wrapper = mount(VarButton, {
    listeners: {
      click: onClick,
      touchstart: onTouchstart,
    },
  })

  await trigger(wrapper, 'click')
  await trigger(wrapper, 'touchstart')
  expect(onClick).toHaveBeenCalledTimes(1)
  expect(onTouchstart).toHaveBeenCalledTimes(1)
  wrapper.destroy()
})

test('test button disabled', async () => {
  const onClick = jest.fn()
  const onTouchstart = jest.fn()

  const wrapper = mount(VarButton, {
    listeners: {
      click: onClick,
      touchstart: onTouchstart,
    },
    propsData: {
      disabled: true,
    },
  })

  await trigger(wrapper, 'click')
  await trigger(wrapper, 'touchstart')
  expect(wrapper.classes()).toContain('var-button--disabled')
  expect(onClick).toHaveBeenCalledTimes(0)
  expect(onTouchstart).toHaveBeenCalledTimes(0)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test button loading', async () => {
  const onClick = jest.fn()
  const onTouchstart = jest.fn()

  const wrapper = mount(VarButton, {
    listeners: {
      click: onClick,
      touchstart: onTouchstart,
    },
    propsData: {
      loading: true,
    },
  })

  await trigger(wrapper, 'click')
  await trigger(wrapper, 'touchstart')
  expect(wrapper.find('.var-loading').exists()).toBeTruthy()
  expect(onClick).toHaveBeenCalledTimes(0)
  expect(onTouchstart).toHaveBeenCalledTimes(0)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test button auto-loading', async () => {
  const onClick = () => {
    return new Promise((resolve) => {
      setTimeout(resolve, 100)
    })
  }

  const onTouchstart = () => {
    return new Promise((resolve) => {
      setTimeout(resolve, 100)
    })
  }

  const wrapper = mount(VarButton, {
    listeners: {
      click: onClick,
      touchstart: onTouchstart,
    },
    propsData: {
      autoLoading: true,
      ripple: false,
    },
  })

  await trigger(wrapper, 'click')
  expect(wrapper.find('.var-loading').exists()).toBeTruthy()
  await delay(100)
  expect(wrapper.find('.var-loading').exists()).toBeFalsy()

  await trigger(wrapper, 'touchstart')
  expect(wrapper.find('.var-loading').exists()).toBeTruthy()
  await delay(100)
  expect(wrapper.find('.var-loading').exists()).toBeFalsy()

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test button color & textColor', () => {
  const wrapper = mount(VarButton, {
    propsData: {
      color: '#000',
      textColor: '#000',
    },
  })
  expect(wrapper.attributes('style')).toMatch('color: rgb(0, 0, 0)')
  expect(wrapper.attributes('style')).toMatch('background: rgb(0, 0, 0)')
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test button styles', () => {
  const wrapper = mount(VarButton, {
    propsData: {
      size: 'large',
      round: true,
      text: true,
      block: true,
    },
    scopedSlots: {
      default: '<div>button text</div>',
    },
  })
  expect(wrapper.classes()).toContain('var-button--round')
  expect(wrapper.classes()).toContain('var-button--text')
  expect(wrapper.classes()).toContain('var-button--block')
  expect(wrapper.classes()).toContain('var-button--large')
  expect(wrapper.text()).toBe('button text')
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})
