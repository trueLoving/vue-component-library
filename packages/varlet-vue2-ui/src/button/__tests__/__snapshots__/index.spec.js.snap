// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button auto-loading 1`] = `
"<button class=\\"var-button var--box var-button--normal var--inline-flex var-button--default var-elevation--2\\">
  <!---->
  <div class=\\"var-button__content\\"></div>
</button>"
`;

exports[`test button color & textColor 1`] = `
"<button class=\\"var-button var--box var-button--normal var--inline-flex var-button--default var-elevation--2\\" style=\\"color: rgb(0, 0, 0); background: rgb(0, 0, 0);\\">
  <!---->
  <div class=\\"var-button__content\\"></div>
</button>"
`;

exports[`test button disabled 1`] = `
"<button disabled=\\"disabled\\" class=\\"var-button var--box var-button--normal var--inline-flex var-button--disabled var-button--default var-elevation--2\\">
  <!---->
  <div class=\\"var-button__content\\"></div>
</button>"
`;

exports[`test button example 1`] = `
"<div class=\\"var-button-example\\">
  <div class=\\"app-type\\">主题色按钮</div> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--default var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">默认按钮</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--primary var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">主要按钮</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--info var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">信息按钮</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--success var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">成功按钮</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--warning var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">警告按钮</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--danger var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">危险按钮</div>
  </button>
  <div class=\\"app-type\\">文字按钮</div> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-primary var-button--text var-button--outline\\">
    <!---->
    <div class=\\"var-button__content\\">外边框按钮</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-primary var-button--text\\">
    <!---->
    <div class=\\"var-button__content\\">纯文字按钮</div>
  </button>
  <div class=\\"app-type\\">禁用状态</div> <button disabled=\\"disabled\\" class=\\"var-button var--box var-button--normal var--inline-flex var-button--disabled var-button--default var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">禁用状态</div>
  </button> <button disabled=\\"disabled\\" class=\\"var-button var--box var-button--normal var--inline-flex var-button--disabled var-button--text-default var-button--text var-button--text-disabled var-button--outline\\">
    <!---->
    <div class=\\"var-button__content\\">禁用状态</div>
  </button> <button disabled=\\"disabled\\" class=\\"var-button var--box var-button--normal var--inline-flex var-button--disabled var-button--text-default var-button--text var-button--text-disabled\\">
    <!---->
    <div class=\\"var-button__content\\">禁用状态</div>
  </button>
  <div class=\\"app-type\\">加载状态</div> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--primary var-elevation--2\\">
    <div class=\\"var-loading var-button__loading\\" var-button-cover=\\"\\">
      <!---->
      <div class=\\"var--box var-loading__body\\">
        <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 24px; height: 24px; color: currentColor;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
    </div>
    <div class=\\"var-button__content var-button--hidden\\">加载状态</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--info var-elevation--2\\">
    <div class=\\"var-loading var-button__loading\\" var-button-cover=\\"\\">
      <!---->
      <div class=\\"var--box var-loading__body\\">
        <!---->
        <!---->
        <!---->
        <div class=\\"var-loading__rect var-loading__rect-normal\\">
          <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__rect-item var-loading__rect-item-normal\\" style=\\"background-color: currentColor;\\"></div>
        </div>
        <!---->
        <!---->
      </div>
    </div>
    <div class=\\"var-button__content var-button--hidden\\">加载状态</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--success var-elevation--2\\">
    <div class=\\"var-loading var-button__loading\\" var-button-cover=\\"\\">
      <!---->
      <div class=\\"var--box var-loading__body\\">
        <!---->
        <!---->
        <!---->
        <!---->
        <div class=\\"var-loading__disappear var-loading__disappear-normal\\">
          <div class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__disappear-item var-loading__disappear-item-normal\\" style=\\"background-color: currentColor;\\"></div>
        </div>
        <!---->
      </div>
    </div>
    <div class=\\"var-button__content var-button--hidden\\">加载状态</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--danger var-elevation--2\\">
    <div class=\\"var-loading var-button__loading\\" var-button-cover=\\"\\">
      <!---->
      <div class=\\"var--box var-loading__body\\">
        <!---->
        <!---->
        <div class=\\"var-loading__cube var-loading__cube-normal\\">
          <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__cube-item var-loading__cube-item-normal\\" style=\\"background-color: currentColor;\\"></div>
        </div>
        <!---->
        <!---->
        <!---->
      </div>
    </div>
    <div class=\\"var-button__content var-button--hidden\\">加载状态</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--warning var-elevation--2\\">
    <div class=\\"var-loading var-button__loading\\" var-button-cover=\\"\\">
      <!---->
      <div class=\\"var--box var-loading__body\\">
        <!---->
        <div class=\\"var-loading__wave var-loading__wave-normal\\">
          <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: currentColor;\\"></div>
          <div class=\\"var-loading__wave-item var-loading__wave-item-normal\\" style=\\"background-color: currentColor;\\"></div>
        </div>
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
    </div>
    <div class=\\"var-button__content var-button--hidden\\">加载状态</div>
  </button>
  <div class=\\"app-type\\">按钮尺寸</div> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--primary var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">常规按钮</div>
  </button> <button class=\\"var-button var--box var-button--small var--inline-flex var-button--success var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">小型按钮</div>
  </button> <button class=\\"var-button var--box var-button--mini var--inline-flex var-button--warning var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">迷你按钮</div>
  </button> <button class=\\"var-button var--box var-button--large var--inline-flex var-button--danger var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">大型按钮</div>
  </button>
  <div class=\\"app-type\\">块级按钮</div> <button class=\\"var-button var--box var-button--normal var--flex var-button--block var-button--primary var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">块级按钮</div>
  </button>
  <div class=\\"app-type\\">自定义颜色</div> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--default var-elevation--2\\" style=\\"color: rgb(238, 238, 238); background: rgb(105, 219, 170);\\">
    <!---->
    <div class=\\"var-button__content\\">背景/文字</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--default var-elevation--2\\" style=\\"color: rgb(255, 255, 255);\\">
    <!---->
    <div class=\\"var-button__content\\">
      使用渐变
    </div>
  </button>
  <div class=\\"app-type\\">圆形按钮</div> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--primary var-elevation--2 var-button--round\\">
    <!---->
    <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-plus\\" style=\\"transition: transform 0ms;\\"></i></div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--info var-elevation--2 var-button--round\\">
    <!---->
    <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-information\\" style=\\"transition: transform 0ms;\\"></i></div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--success var-elevation--2 var-button--round\\">
    <!---->
    <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-check\\" style=\\"transition: transform 0ms;\\"></i></div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--warning var-elevation--2 var-button--round\\">
    <!---->
    <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-warning\\" style=\\"transition: transform 0ms;\\"></i></div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--danger var-elevation--2 var-button--round\\">
    <!---->
    <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-window-close\\" style=\\"transition: transform 0ms;\\"></i></div>
  </button>
  <div class=\\"app-type\\">注册事件</div> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--success var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">点击</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--success var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">触摸</div>
  </button> <button class=\\"var-button var--box var-button--normal var--inline-flex var-button--success var-elevation--2\\">
    <!---->
    <div class=\\"var-button__content\\">自动加载</div>
  </button>
</div>"
`;

exports[`test button loading 1`] = `
"<button class=\\"var-button var--box var-button--normal var--inline-flex var-button--default var-elevation--2\\">
  <div class=\\"var-loading var-button__loading\\" var-button-cover=\\"\\">
    <!---->
    <div class=\\"var--box var-loading__body\\">
      <div class=\\"var-loading__circle\\"><span class=\\"var-loading__circle-block\\" style=\\"width: 24px; height: 24px; color: currentColor;\\"><svg viewBox=\\"25 25 50 50\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle></svg></span></div>
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
  </div>
  <div class=\\"var-button__content var-button--hidden\\"></div>
</button>"
`;

exports[`test button styles 1`] = `
"<button class=\\"var-button var--box var-button--large var--flex var-button--block var-button--text-default var-button--text var-button--round\\">
  <!---->
  <div class=\\"var-button__content\\">
    <div>button text</div>
  </div>
</button>"
`;
